frame/support/procedural/src/construct_runtime/expand/outer_enums.rs

```diff
@@ -0,0 +1,284 @@
+		if enum_ty == OuterEnumType::Event { quote!(Clone, PartialEq, Eq,) } else { quote!() };
+
+	// Implementation specific for errors.
+	let error_custom_impl = generate_error_impl(scrate, enum_ty);
+
+	Ok(quote! {
+		#( #query_enum_part_macros )*
+
+		#[derive(
+			#event_custom_derives
+			#scrate::codec::Encode,
+			#scrate::codec::Decode,
+			#scrate::scale_info::TypeInfo,
+			#scrate::RuntimeDebug,
+		)]
+		#[allow(non_camel_case_types)]
+		pub enum #enum_name_ident {
+			#enum_variants
+		}
+
+		#enum_conversions
+
+		#error_custom_impl
+	})
+}
+
+fn expand_enum_variant(
+	runtime: &Ident,
+	pallet: &Pallet,
+	index: u8,
+	instance: Option<&Ident>,
+	generics: &Generics,
+	enum_ty: OuterEnumType,
+) -> TokenStream {
+	let path = &pallet.path;
+	let variant_name = &pallet.name;
+	let part_is_generic = !generics.params.is_empty();
+	let attr = pallet.cfg_pattern.iter().fold(TokenStream::new(), |acc, pattern| {
+		let attr = TokenStream::from_str(&format!("#[cfg({})]", pattern.original()))
+			.expect("was successfully parsed before; qed");
+		quote! {
+			#acc
+			#attr
+		}
+	});
+
+	match instance {
+		Some(inst) if part_is_generic => quote! {
+			#attr
+			#[codec(index = #index)]
+			#variant_name(#path::#enum_ty<#runtime, #path::#inst>),
+		},
+		Some(inst) => quote! {
+			#attr
+			#[codec(index = #index)]
+			#variant_name(#path::#enum_ty<#path::#inst>),
+		},
+		None if part_is_generic => quote! {
+			#attr
+			#[codec(index = #index)]
+			#variant_name(#path::#enum_ty<#runtime>),
+		},
+		None => quote! {
+			#attr
+			#[codec(index = #index)]
+			#variant_name(#path::#enum_ty),
+		},
+	}
+}
+
+fn expand_enum_conversion(
+	scrate: &TokenStream,
+	pallet: &Pallet,
+	pallet_enum: &TokenStream,
+	enum_name_ident: &Ident,
+) -> TokenStream {
+	let variant_name = &pallet.name;
+	let attr = pallet.cfg_pattern.iter().fold(TokenStream::new(), |acc, pattern| {
+		let attr = TokenStream::from_str(&format!("#[cfg({})]", pattern.original()))
+			.expect("was successfully parsed before; qed");
+		quote! {
+			#acc
+			#attr
+		}
+	});
+
+	quote! {
+		#attr
+		impl From<#pallet_enum> for #enum_name_ident {
+			fn from(x: #pallet_enum) -> Self {
+				#enum_name_ident
+				::#variant_name(x)
+			}
+		}
+		#attr
+		impl TryInto<#pallet_enum> for #enum_name_ident {
+			type Error = ();
+
+			fn try_into(self) -> #scrate::sp_std::result::Result<#pallet_enum, Self::Error> {
+				match self {
+					Self::#variant_name(evt) => Ok(evt),
+					_ => Err(()),
+				}
+			}
+		}
+	}
+}
+
+fn generate_error_impl(scrate: &TokenStream, enum_ty: OuterEnumType) -> TokenStream {
+	// Implementation is specific to `Error`s.
+	if enum_ty == OuterEnumType::Event {
+		return quote! {}
+	}
+
+	let enum_name_ident = Ident::new(enum_ty.struct_name(), Span::call_site());
+
+	quote! {
+		/// Optionally convert the `DispatchError` into the `RuntimeError`.
+		///
+		/// Returns `Some` if the error matches the `DispatchError::Module` variant, otherwise `None`.
```

Comment:
The docs need moving down a line to be on the actual function :)

Reply:
Ops! Thanks for catching this! 🙏